linux/runtime.go

```diff
@@ -409,9 +401,37 @@ func (r *Runtime) loadTasks(ctx context.Context, ns string) ([]*Task, error) {
 			log.G(ctx).WithError(err).Error("loading task type")
 			continue
 		}
-		o = append(o, t)
+		// TODO: need to add the tasks to the monitor
+		if err := r.tasks.AddWithNamespace(ns, t); err != nil {
+			return err
+		}
+
+		// Start a routine to wait upon an unexpected exit of the shim. So as
+		// to clean up after it. If the wait succeeds then no cleanup should
+		// be necessary.
+		go func() {
+			ctx = namespaces.WithNamespace(context.Background(), ns)
+			for {
+				_, err := s.Wait(ctx, &shim.WaitRequest{id})
```

Comment:
Instead of having a blocking connection for every shim, could we change ttprc to have a shutdown hook.  `ttrpc.OnShutdown(func() {})` that any call that gets an `ErrClosed` runs the shutdown function to cleanup the connection? @stevvooe

Reply:
I was thinking a goroutine + connection for every shim is heavy.

Reply:
Currently it's the only way I can see, unless we go with @crosbymichael suggestion, which would definitely be better on resources usage

Reply:
Looking closer, an `OnShutdown method in ttrpc would help here, assuming we are tying everything to the lifecycle of the client, and not the connection. If we want better control over the connection lifecyle, we'll have to introduce a pool. In this case, we are clearly using the client lifecycle, so this change would work well.